pkg/common/naming.go

```diff
@@ -43,6 +43,20 @@ func EndpointHostname(workspaceId, endpointName string, endpointPort int, routin
 	return fmt.Sprintf("%s.%s", hostname, routingSuffix)
 }
 
+ // WorkspaceHostname evaluates a single hostname for a workspace, and should be used for routing 
+ // when endpoints are distinguished by path rules
+func WorkspaceHostname(workspaceId, routingSuffix string) string {
+	hostname := workspaceId
+	if len(hostname) > 63 {
+		hostname = strings.TrimSuffix(hostname[:63], "-")
+	}
+	return fmt.Sprintf("%s.%s", hostname, routingSuffix)
+}
+
+func EndpointPath(endpointName string) string {
+	return "/" + endpointName + "/"
```

Comment:
In testing on `crc` I found a few issues with our approach to paths:
- When we suffix the path with `/`, that exact string must be used to access the endpoint. This is an issue because the ideURL ends up without the trailing `/` so clicking the ideURL takes you to the OpenShift "application not available" page.
- The workspace never enters a `Running` state -- it's stuck in `Starting`. I believe (but haven't verified) that this is because we use `ideURL` in `checkServerStatus()`: 
    ```go
    ideUrl := workspace.Status.IdeUrl
    healthz, err := url.Parse(ideUrl)
    healthz.Path = healthz.Path + "healthz"
    ```
    since I'm seeing `IdeURL` as e.g. https://workspace99cfef48974c4fef.apps-crc.testing/theia it would appear that we're trying to ping `https://workspace99cfef48974c4fef.apps-crc.testing/theiahealthz`, which returns 503 instead of a 4xx that we're expecting. This causes the readiness check to fail at the end.

Reply:
ideUrl was with trailing `/`... probably it went after Path.join stuff
Interesting that theia is not loaded when you access it without a trailing slash, seems Theia needs the right redirect:
without trailing slash
![Screenshot_20201201_142351](https://user-images.githubusercontent.com/5887312/100740201-fcafc280-33e0-11eb-974d-c04fa2603f37.png)
with trailing slash
![Screenshot_20201201_142340](https://user-images.githubusercontent.com/5887312/100740205-fe798600-33e0-11eb-8631-3b1c406fbb78.png)

So, I'll make sure that endpoints have trailing slash if it's path rule, and it should be possible to specify endpoint path without a slash.

Reply:
Should be fixed with https://github.com/devfile/devworkspace-operator/pull/201/commits/db048ad7543389f7c9d57e4b279ffd418b1e5e6f